lib/ui/dart_runtime_hooks.cc

```diff
@@ -199,6 +202,42 @@ void Logger_PrintString(Dart_NativeArguments args) {
   }
 }
 
+void SaveCompilationTrace(Dart_NativeArguments args) {
+  const char* filePath = NULL;
+  {
+    Dart_Handle result = Dart_StringToCString(
+        Dart_GetNativeArgument(args, 0), &filePath);
+    if (Dart_IsError(result)) {
+      Dart_PropagateError(result);
+      return;
+    }
+  }
+
+  uint8_t* buffer = nullptr;
+  intptr_t size = 0;
+  Dart_Handle result = Dart_SaveCompilationTrace(&buffer, &size);
+  if (Dart_IsError(result)) {
+    FXL_LOG(ERROR) << "Failed to save trace: " << Dart_GetError(result);
+    return;
+  }
+
+  auto fd = fml::UniqueFD{
+      FML_HANDLE_EINTR(::open(filePath, O_CREAT|O_WRONLY|O_TRUNC, 0660))};
```

Comment:
This file in unlikely to compile or run on Windows but is built on that platform. I suggest you return a typed unsigned integer list from this method and write the file to a destination using `dart:io` APIs in the wrappers in `natives.dart`.

Reply:
It should work on Windows because Windows doesn't have EINTR problem, and so on Windows we define FML_HANDLE_EINTR(::open(...)) as simply ::open(...). Having said this, let me know if you still want this to use dart::io instead.

Reply:
Yes please. I would prefer the engine be as simple as possible and just manage trace acquisition. How the trace is stored/transferred doesn't concern the engine.

Reply:
Done.